flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsStateBackend.java

```diff
@@ -97,6 +100,27 @@ public FsStateBackend(String checkpointDataUri) throws IOException {
 	 *
 	 * @param checkpointDataUri The URI describing the filesystem (scheme and optionally authority),
 	 *                          and the path to the checkpoint data directory.
+	 * @param asynchronousSnapshots Switch to enable asynchronous snapshots.
+	 *
+	 * @throws IOException Thrown, if no file system can be found for the scheme in the URI.
+	 */
+	public FsStateBackend(String checkpointDataUri, boolean asynchronousSnapshots) throws IOException {
```

Comment:
I agree that the explosion is not nice, and it is certainly easy to change that. I just have a preference to initialize the configuration as eager and `final` as possible